File: source/control/detumbling.c
void Detumbling(MatrixMath_s* commDipole, MatrixMath_s bMeas, MatrixMath_s bDeriv, double ctrlgains_bdot) {
  double temp_1e9 = pow(10, -9);
  double temp = ctrlgains_bdot * temp_1e9;
  double r = 0;

  multiplyScalar(temp * (-1), &bDeriv);
  multiplyScalar(temp_1e9, &bMeas);
  r = pow(vectorNorm(&bMeas), 2);

  multiplyScalar(1/r, &bDeriv);
  (*commDipole) = bDeriv;
}

File: Matlab/Control/Detumbling.m
function commDipole = Detumbling(bMeas, bDeriv)
% Calculating desired control dipole for Detumbling mode [Am2]
% desired control torque is expressed in satellite frame
% control law is based on B-Dot algorithm
% underactuation of the coils is included

global ctrlgains;
k = ctrlgains.bdot;

commDipole = -k*bDeriv*1e-9 / vectorNorm(bMeas*1e-9)^2;